### PR TITLE
Return error information on HTTP block request

### PR DESCRIPTION
**Pull request by gsambrotta**
Currently, the HTTP block handle error in the code but it does not return any information about it.
If a GET/POST/PUT/... request is made and an error is raised, the data object in the flow is `null`

With this PR the data object in the flow, will now return the error information.
It also contain a very handy `hasError: true` property so it can be easily check for error or not and potentially display error messages in the flow.

Here two screenshots of how the result will be:
image
image

And here is a flow that can be use to test this PR: [Kendraio Harvest Time](https://app.kendra.io/harvest-timesheet/harvestTimesheet)

Linear Issue [K-361](https://linear.app/kendraio/issue/K-361/kendraio-time-flow)

### CODE REVIEW

**Comment by lukestanley** [comment]
This seems good, nice work! It needs tests updating though:
see: `cypress/e2e/http_block_page_following.ts`
@gsambrotta

File: cypress/e2e/http_block.ts
Comment (gsambrotta) [inline cypress/e2e/http_block.ts:6]: Unfortunately changing the file name made code review a bit more difficult.
To help here some info: 
- I divided the test in two test suits: a normal http request and paginated request
- The new code is inside the first test suit: `HTTP Block Request`

**Review by lukestanley — CHANGES_REQUESTED** [review]
We need to see that the error code and body can be used too. I don't think this does that, right?
@gsambrotta

**Comment by gsambrotta** [comment]
> We need to see that the error code and body can be used too. I don't think this does that, right? @gsambrotta

@lukestanley It does in the `onError` config. https://github.com/kendraio/kendraio-app/pull/393/files#diff-5da5ddc4e5c0ad1ea378ff3988f2537b9975bfc9c90fb9ba8d31a75ee323858eR74
Here I use `data.error.error_message` to display the message. Obviously it can be use also `data.error.statusCode`

For some reasons it does not get display in the Cypress view, so I cannot `cy.get` like I do with the error static message.
I think the data get populated after the card is created and it doesn't display the message.

Do you think is not clear like it is now? Is needed to check the existence of the error message or status code?

File: docs/workflow/blocks/http.rst
Comment (CodeKrakken) [inline docs/workflow/blocks/http.rst:55]: Better English would be - 

"valueGetter" can only get data from the context.

**Review by CodeKrakken — CHANGES_REQUESTED** [review]
Just that one syntax comment. Code looks clear and all tests are passing.

File: docs/workflow/blocks/http.rst
Comment (gsambrotta) [reply docs/workflow/blocks/http.rst:55]: Thanks!

File: docs/workflow/blocks/http.rst
Comment (lukestanley) [inline docs/workflow/blocks/http.rst:92]: What's going on here? @gsambrotta

Comment (gsambrotta) [reply docs/workflow/blocks/http.rst:92]: Ah, thanks! I think I was typing somewhere else, and it accidentally typed on that line.

File: docs/workflow/blocks/http.rst
Comment (lukestanley) [inline docs/workflow/blocks/http.rst:36]:
```suggestion
- **onError** - Define an array of blocks to show when there is an error processing the HTTP request. 
```

File: docs/workflow/blocks/http.rst
Comment (lukestanley) [inline docs/workflow/blocks/http.rst:92]:
```suggestion
      "method": "post",
```

File: docs/workflow/blocks/http.rst
Comment (lukestanley) [inline docs/workflow/blocks/http.rst:86]: This part needs an example probably.
I tried to improve it as is though:
```suggestion
Caution: if the custom header value is a string, header values must use two types of quotes: double quotes and single quotes.
```

Comment (gsambrotta) [reply docs/workflow/blocks/http.rst:86]: Yeah, you are right, is not clear, but I don't know how to explain it. 
I am not sure about your suggestion, neither, though.
Why did you write `custom header`? This tip (double+single quotes) is valid for any headers, custom or not. Or when we say "custom header" we mean "non-dynamic value"?

There is an example, is right after this text, in the http block example. https://github.com/kendraio/kendraio-app/pull/393/files#diff-544912a428316debac6dc397877aae4d88c8bdac4836974e3caa2c8389ffc96dR98

But yeah, maybe we can be more explicit and add this line `"payload": "'grant_type=client_credentials'"` in the explanation.

Something like:
> Caution: if the header value is a string, the header value must use two types of quotes: double quotes and single quotes like in the following example: `"payload": "'grant_type=client_credentials'"`

What do you think @lukestanley ?

Actually, I have no idea if it should be in double + single quotes in the case the value is a number or boolean. 
But I don't think we will ever have a header that is not either a string or a dynamic value

Comment (lukestanley) [reply docs/workflow/blocks/http.rst:86]: Actually I'd replace this:
`For advanced use cases, the payload can be constructed using a JMES Path expression.
Custom headers can also be specified using JMES Path expressions:
Custom headers can also be specified using JMESPath expressions.`
With:

`JMESPath expressions can be used to dynamically set header values.` Rather than the duplication. This would be more clear.

And thanks for pointing out that there is an example below. @gsambrotta

Comment (gsambrotta) [reply docs/workflow/blocks/http.rst:86]: Good catch to avoid repetition. I think it still good to mention the payload tho. 
What about this text for the whole description:
```
JMESPath expressions can be used to dynamically set header and payload values.
Caution: if the header value is a string, it must use two types of quotes: double and single quotes, like "payload": "'grant_type=client_credentials'".
```
Is this good for you @lukestanley ?

Comment (lukestanley) [reply docs/workflow/blocks/http.rst:86]: I thumbed up suggested text last week. So a commit with that in it is needed, right?  @gsambrotta

**Comment by gsambrotta** [comment]
@lukestanley I committed the text as we agreed. Would you please give another check?

**Review by CodeKrakken — COMMENTED** [review]
@gsambrotta That all looks good to me.

**Comment by gsambrotta** [comment]
> @gsambrotta That all looks good to me.

@CodeKrakken If so, can you please approve it?

**Comment by CodeKrakken** [comment]
> > @gsambrotta That all looks good to me.
> 
> @CodeKrakken If so, can you please approve it?

Sorry - was waiting for @lukestanley to review it.

**Comment by gsambrotta** [comment]
@CodeKrakken ahh you are right. I thought we configure to have at least 1 approval but apparently an approval from anyone that has done the review is needed.

**Review by lukestanley — APPROVED** [review]
Looks good! Well done! @gsambrotta

**Pull request merged**
